libraries/AC_AttitudeControl/AC_AttitudeControl_Logging.cpp

```diff
@@ -0,0 +1,82 @@
+#include <AP_Logger/AP_Logger_config.h>
+
+#if HAL_LOGGING_ENABLED
+
+#include "AC_AttitudeControl.h"
+#include "AC_PosControl.h"
+#include <AP_Logger/AP_Logger.h>
+#include <AP_Scheduler/AP_Scheduler.h>
+#include "LogStructure.h"
+
+// Write an ANG packet
+void AC_AttitudeControl::Write_Attitude() const
+{
+    Vector3f targets = get_att_target_euler_rad() * RAD_TO_DEG;
+    // Retrieve quaternion body attitude
+    Quaternion attitude_body;
+    _ahrs.get_quat_body_to_ned(attitude_body);
+    // calculate the attitude body euler angles
+    Vector3f attitude_body_angles;
+    attitude_body.to_euler(attitude_body_angles);
+    attitude_body_angles *= RAD_TO_DEG;
+
+    const struct log_Ang pkt{
+        LOG_PACKET_HEADER_INIT(LOG_ANG_MSG),
+        time_us         : AP::scheduler().get_loop_start_time_us(),
+        control_roll    : targets.x,
+        roll            : attitude_body_angles.x,
+        control_pitch   : targets.y,
+        pitch           : attitude_body_angles.y,
+        control_yaw     : wrap_360(targets.z),
+        yaw             : wrap_360(attitude_body_angles.z),
+        sensor_dt       : AP::scheduler().get_last_loop_time_s()
+    };
+    AP::logger().WriteBlock(&pkt, sizeof(pkt));
+}
+
+// Write a rate packet
+void AC_AttitudeControl::Write_Rate(const AC_PosControl &pos_control) const
```

Comment:
Taking the position controller here is a little odd.  Perhaps we should pass in the targets instead like we do for `ATT`?

... it's a bit off that we don't persist the targets within here somehow already...

Reply:
Its better than it was - took 3 arguments. Regardless this is code that has already been approved and merged so I don't propose to dork with it here.